Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -232,12 +290,25 @@ static CFStringRef _CFStringFromLocalizedString(IDWriteLocalizedStrings* localiz
         NSDictionary* attribs = [ts->_attributedString attributesAtIndex:i
                                                    longestEffectiveRange:&attributeRange
                                                                  inRange:{ i, [ts->_attributedString length] - i }];
-        CGFloat fontSize = kCTFontSystemFontSize;
-        UIFont* font = [attribs objectForKey:static_cast<NSString*>(kCTFontAttributeName)];
+
+        CTFontRef font = static_cast<CTFontRef>([attribs objectForKey:static_cast<NSString*>(kCTFontAttributeName)]);
         if (font != nil) {
-            fontSize = [font pointSize];
+            CGFloat fontSize = CTFontGetSize(font);
+            DWRITE_FONT_WEIGHT weight = DWRITE_FONT_WEIGHT_REGULAR;
+            DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+            DWRITE_FONT_STRETCH stretch = DWRITE_FONT_STRETCH_NORMAL;
+            __InitDWriteFontPropertiesFromName(CTFontCopyName(font, kCTFontFullNameKey), &weight, &stretch, &style);
+            CFStringRef name = _DWriteGetFamilyNameForFontName(CTFontCopyName(font, kCTFontFullNameKey));
+            std::vector<wchar_t> fontName(CFStringGetLength(name) + 1);
+            CFStringGetCharacters(name, CFRangeMake(0, fontName.size()), reinterpret_cast<UniChar*>(fontName.data()));
+
+            THROW_IF_FAILED(textLayout->SetFontSize(fontSize, DWRITE_TEXT_RANGE{ attributeRange.location, attributeRange.length }));
```

Comment:
>DWRITE_TEXT_RANGE{ attributeRange.location, attributeRange.length } [](start = 62, length = 67)

const DWRITE_TEXT_RANGE range {...} and then use them.  compiler will likely optimize it the same, but the code would be cleaner.